credentials/credentialset.go

```diff
@@ -77,15 +77,15 @@ func Load(path string) (*CredentialSet, error) {
 
 // Validate compares the given credentials with the spec.
 //
-// This will result in an error only if:
-// - a parameter in the spec is not present in the given set
-// - a parameter in the given set does not match the format required by the spec
+// This will result in an error only when the following conditions are true:
+// - a credential in the spec is not present in the given set
+// - the credential is required
 //
 // It is allowed for spec to specify both an env var and a file. In such case, if
-// the givn set provides either, it will be considered valid.
+// the givesn set provides either, it will be considered valid.
```

Comment:
Nice try. How about this:
```suggestion
// the given set provides either, it will be considered valid.
```